lib/OptimizationMOI/src/OptimizationMOI.jl

```diff
@@ -17,28 +20,36 @@ function SciMLBase.allowsconstraints(opt::Union{MOI.AbstractOptimizer,
     true
 end
 
-struct MOIOptimizationProblem{T, F <: OptimizationFunction, uType, P,
+struct MOIOptimizationCache{T, F <: OptimizationFunction, uType, P, LB, UB, I,
                               JT <: DenseOrSparse{T}, HT <: DenseOrSparse{T},
-                              CHT <: DenseOrSparse{T}} <:
+                              CHT <: DenseOrSparse{T}, S, PR} <:
        MOI.AbstractNLPEvaluator
     f::F
     u0::uType
     p::P
+    lb::LB
+    ub::UB
+    int::I
     J::JT
     H::HT
     cons_H::Vector{CHT}
     lcons::Vector{T}
     ucons::Vector{T}
+    sense::S
+    prob::PR # TODO: this is dangerous since the cache and problem might get out of sync after reinit
```

Comment:
Let's store the cache in the solution object as you suggested below, to make full use of the memory efficiency the abstraction offers? (again this would have to be done in SciMLBase)